From c0cd5bdffb3123ba4c30e106fae0e0db1c857e35 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Tue, 8 Oct 2024 16:00:43 +0200
Subject: [PATCH 1306/1695] plat-stm32mp2: allow BSEC compilation with CM33
 TDCID

Remove CFG_STM32_BSEC3 & CFG_STM32_BSEC_PTA from the list of flags that
are forced to n if Cortex-M is TDCID (flag CFG_STM32_CM33TDCID).
Also put init_debug() function under CFG_STM32_CM33TDCID flag, as BSEC
peripheral cannot be accessed in this case.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f5e5c588e7f29437201e86377604de2b4d2b2fe7 plat-stm32mp2: allow BSEC compilation with CM33 TDCID]
Change-Id: I8629a9881e3931404dd20c84580849867e0118a1
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/409836
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 2 --
 core/arch/arm/plat-stm32mp2/main.c  | 2 ++
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 9295869a8..b67158e26 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -94,8 +94,6 @@ $(call force,CFG_SCMI_SERVER_CLOCK_CONSUMER,n)
 $(call force,CFG_SCMI_SERVER_PD_CONSUMER,n)
 $(call force,CFG_SCMI_SERVER_REGULATOR_CONSUMER,n)
 $(call force,CFG_SCMI_SERVER_RESET_CONSUMER,n)
-$(call force,CFG_STM32_BSEC_PTA,n)
-$(call force,CFG_STM32_BSEC3,n)
 $(call force,CFG_STM32_BSEC_WRITE,n)
 $(call force,CFG_STM32MP_PROVISIONING,n)
 $(call force,CFG_TA_STM32MP_NVMEM,n)
diff --git a/core/arch/arm/plat-stm32mp2/main.c b/core/arch/arm/plat-stm32mp2/main.c
index aec3929c8..77c3fdfec 100644
--- a/core/arch/arm/plat-stm32mp2/main.c
+++ b/core/arch/arm/plat-stm32mp2/main.c
@@ -178,6 +178,7 @@ void plat_bsec_get_static_cfg(struct stm32_bsec_static_cfg *cfg)
 	cfg->max_id = STM32MP2_OTP_MAX_ID;
 }
 
+#ifndef CFG_STM32_CM33TDCID
 #define BSEC3_DEBUG_ALL		GENMASK_32(11, 1)
 static TEE_Result init_debug(void)
 {
@@ -232,6 +233,7 @@ static TEE_Result init_debug(void)
 	return res;
 }
 early_init_late(init_debug);
+#endif /* !CFG_STM32_CM33TDCID */
 #endif /* CFG_STM32_BSEC3 */
 
 #ifdef CFG_STM32_CPU_OPP
-- 
2.47.2

